### PR TITLE
Switch to Guzzle 2.x

### PR DESCRIPTION
**Pull request by shadowhand**
Refs #13

### CODE REVIEW

**Comment by Nyholm** [comment]
Thank you for this PR. I dont think this is needed. Guzzlehttp/psr7:2.0 has factories by itself. Or am I missing somthing?

**Comment by shadowhand** [comment]
Oh, well that is interesting: https://github.com/guzzle/psr7/blob/2.0.0/src/HttpFactory.php

Should we deprecate this package and recommend `guzzlehttp/psr7 ^2.0` instead?

**Comment by Nyholm** [comment]
Yeah. It make sense. 

The readme can state that this is for Guzzlehttp/psr7:1 only. I guess it will be a few more years before people stop using Guzzlehttp/psr7:1

File: composer.json [hunk 1]
```diff
@@ -18,12 +18,13 @@
         "psr/http-factory-implementation": "^1.0"
     },
     "require": {
+        "php": ">=7.3",
         "psr/http-factory": "^1.0",
-        "guzzlehttp/psr7": "^1.4.2||^2.0"
```
Comment (GrahamCampbell) [inline composer.json:22]: why not just allow `guzzlehttp/psr7": "^1.7 || ^2.0`

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

**Pull request closed without merging**